### PR TITLE
python: Deprecate symbols that are planned for removal in 8.0.0

### PR DESCRIPTION
**Pull request by da-tanabe**
* Deprecate symbols that are planned for removal in 8.0.0
* Fix regressions in `dazl ls` and improve its performance.

This PR will be easier to review when ignoring whitespace changes; aside from fixing the `dazl ls` regression, the PR is simply adding `warnings.warn` and `warnings.ignore_warnings` where deprecated code calls other deprecated code (Python doesn't automatically suppress warnings in this context so logs would be flooded with deprecation warnings without the extra ignores).

### CODE REVIEW

File: python/dazl/pretty/table/fmt_pretty.py [hunk 1]
```diff
@@ -47,14 +46,69 @@ def render(
             # compact representation here
             yield ''
 
-            renderer = _TemplateEntryRenderer(store, name, parties)
-            renderer.measure(entries)
+            tt = con(name)
+            context = Context(ArrayStringMapper(), lookup)
+            header_row = ['', '#cid', *(expand_record_field_names(context, Type.Con(name, ())))]
+
+            with LOG.debug_timed("Render entries as strings"):
```
Comment (mschaef-da) [inline python/dazl/pretty/table/fmt_pretty.py:53]: This is a nice pattern.

File: python/dazl/pretty/table/fmt_pretty.py [hunk 2]
```diff
@@ -47,14 +46,69 @@ def render(
...
+            with LOG.debug_timed("Render rows"):
+                header_len = sum(cell_sizes) + (len(cell_sizes) - 1) * len(col_spacer)
+                header = f"{name} ({contract_count(len(entries))}) "
+
+                yield header + '-' * max(3, header_len - len(header))
+                for row in rows:
+                    yield col_spacer.join(cell.ljust(cell_size) for cell_size, cell in zip(cell_sizes, row))
+
+
+def contract_count(c: int) -> str:
```
Comment (mschaef-da) [inline python/dazl/pretty/table/fmt_pretty.py:81]: Thank you. Pet Peeve of mine when systems get this wrong.

File: python/dazl/protocols/v0/json_ser_command.py [hunk 3]
```diff
@@ -43,20 +41,21 @@ def to_api_datetime(obj):
...
 class JsonSerializer(AbstractSerializer):
 
-    def mapper(self):
-        from ...values.json import ENCODER
-        return ENCODER
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        warnings.warn(
+            'dazl.protocols.v0.json_ser_command.JsonSerializer is deprecated; '
+            'there is no replacement.', DeprecationWarning, stacklevel=2)
```
Comment (mschaef-da) [inline python/dazl/protocols/v0/json_ser_command.py:56]: I understand why stack frame paramaters are needed, but I've never liked them at all. (No change reqeuested, just complaining.)

File: python/dazl/protocols/v1/grpc.py [hunk 4]
```diff
@@ -215,54 +224,61 @@ def __init__(self, connection: 'GRPCv1Connection', ledger_id: str):
...
-    for package_id, archive_payload in metadatas_pb.sorted_archives.items():
-        store.register_all(parse_daml_metadata_pb(package_id, archive_payload))
-
-    LOG.debug("grpc_package_sync ended.")
+    warnings.warn(
+        'grpc_package_sync is deprecated; there is no replacement',
+        DeprecationWarning, stacklevel=2)
+
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore', DeprecationWarning)
```
Comment (mschaef-da) [inline python/dazl/protocols/v1/grpc.py:252]: I like this mechanism too, and did not know Python supported it. Very nice.

**Review by mschaef-da — APPROVED** [review]
Looks good. (And I learned a few things about Python in the process of the review.)

File: python/dazl/pretty/table/fmt_pretty.py (hunk 2 above)
Comment (da-tanabe) [reply python/dazl/pretty/table/fmt_pretty.py:81]: Hah…me too. :-)

File: python/dazl/protocols/v0/json_ser_command.py (hunk 3 above)
Comment (da-tanabe) [reply python/dazl/protocols/v0/json_ser_command.py:56]: yeah I agree…this is incredibly annoying.

**Pull request merged**